Mathlib/Algebra/Divisibility/Ring.lean

```diff
@@ -0,0 +1,81 @@
+/-
+Copyright (c) 2023 Oliver Nash. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Oliver Nash
+-/
+import Mathlib.Algebra.Divisibility.Basic
+import Mathlib.Data.Nat.Choose.Sum
+import Mathlib.Algebra.BigOperators.Ring
+import Mathlib.Tactic.Linarith
+
+/-!
+# Lemmas about divisibility in rings
+-/
+
+variable {R : Type*} {x y : R}
+
+lemma dvd_nsmul_of_dvd [Semiring R] {n : ℕ} (h : x ∣ y) : x ∣ n • y :=
+  ⟨n • Classical.choose h, by rw [mul_smul_comm, ← Classical.choose_spec h]⟩
```

Comment:
Using `choose` here feels a bit sloppy; does this or something similar work?
```suggestion
  let ⟨k, hk⟩ := h; ⟨n • k, by rw [mul_smul_comm, ← hk]⟩
```

Reply:
OK done.